lib/agency.rb

```diff
@@ -0,0 +1,39 @@
+class Agency
+  attr_reader :region
+
+  def initialize(data)
+    @data = data
+
+    data.tap do |d|
+      @name = d["Supplier Name"]
```

Comment:
Just a suggestion. Might be worth downcasing and snakecasing the field names in the input agencies csv file so that "Supplier Name" for example will become "supplier_name" here. This will avoid case mismatch errors and cater for missing spaces.

Reply:
It's a good thought but unfortunately we don't generate these CSVs (we're given them as-is) so they need to work with what we're provided